src/sagas/projectsSaga.js

```diff
@@ -0,0 +1,68 @@
+import { call, put, takeLatest, all } from 'redux-saga/effects';
+import { getAllProjects } from '../service/projectsService';
+import {
+  actionTypes
+} from '../components/projects/modules/types';
+import {
+  fetchProjectsSuccess,
+  fetchProjectsFailure,
+  fetchProjectsRequest, 
+} from '../components/projects/modules/actions';
+
+
+function* fetchProjectsSaga(action) {
+  const { page, pageSize, selectedProjectStatus, searchInput } = action.payload;
+
+  try {
+    const response = yield call(
+      getAllProjects,
+      page,
+      pageSize,
+      selectedProjectStatus,
+      searchInput
+    );
+
+    const projects = response.data.projects;
+    const total = response.data.count;
+
+    const formattedData = projects.map((project, index) => {
```

Comment:
can we reduce complexity of this function using just one map, by removing map inside of a map